Paths not trimmed throws exception when path starts with space

Below is the command line I used and accidentally left a space in the beginning when I was trying to call a file directly. You can duplicate it with any file type by leaving a space at the beginning of the quotes.

Environment:
* Windows 10.0.19045 Pro Build 19045
* 48 GB RAM
* Release is 1.6.0

```cmd
validator -quiet " C:\Temp\dotnet_examples\DotNet Examples\BenchmarkRunner\TestObjects\HashFile_Small.xml"
```

Here's the output:

```cmd
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x28 pc=0xa075a5]

goroutine 1 [running]:
github.com/Boeing/config-file-validator/pkg/finder.FileSystemFinder.findOne.func1({0xc0000660c0, 0x57}, {0x0, 0x0}, {0x57?, 0x5c?})
        /github/workspace/pkg/finder/fsfinder.go:122 +0x45
path/filepath.WalkDir({0xc0000660c0, 0x57}, 0xc000087810)
        /usr/local/go/src/path/filepath/path.go:531 +0x9c
github.com/Boeing/config-file-validator/pkg/finder.FileSystemFinder.findOne({{0xc000020500, 0x1, 0x1}, {0xd19540, 0xb, 0xb}, 0xc00001b590, 0xc00001b5c0, 0x0}, {0xc0000660c0, ...})
        /github/workspace/pkg/finder/fsfinder.go:119 +0x198
github.com/Boeing/config-file-validator/pkg/finder.FileSystemFinder.Find({{0xc000020500, 0x1, 0x1}, {0xd19540, 0xb, 0xb}, 0xc00001b590, 0xc00001b5c0, 0x0})
        /github/workspace/pkg/finder/fsfinder.go:81 +0x18b
github.com/Boeing/config-file-validator/pkg/cli.CLI.Run({{0xb1db40?, 0xc000065ae0?}, {0xb1dba0?, 0xd9e580?}})
        /github/workspace/pkg/cli/cli.go:82 +0x56
main.mainInit()
        /github/workspace/cmd/validator/validator.go:225 +0x6e7
main.main()
        /github/workspace/cmd/validator/validator.go:234 +0x13
```

If you remove the space at the beginning of the quoted path it runs correctly. Not a major issue since there is a workaround.

--- a/pkg/finder/finder_test.go
+++ b/pkg/finder/finder_test.go
@@ -249,10 +249,11 @@ func Test_FileFinderBadPath(t *testing.T) {
 	}
 }
 
-func Test_FileFinderPathWithLeadingAndTrailingWhitespaces(t *testing.T) {
+func Test_FileFinderPathWithWhitespaces(t *testing.T) {
 	tests := []struct {
-		name string
-		path string
+		name      string
+		path      string
+		expectErr bool
 	}{
 		{
 			name: "no whitespace",
@@ -270,6 +271,26 @@ func Test_FileFinderPathWithLeadingAndTrailingWhitespaces(t *testing.T) {
 			name: "leading and trailing whitespace",
 			path: "  ../../test/fixtures/subdir  ",
 		},
+		{
+			name:      "whitespace in middle of path",
+			path:      "../../test/  fixtures  /subdir",
+			expectErr: true,
+		},
+		{
+			name:      "leading whitespace + whitespace in middle of path",
+			path:      "  ../../test/  fixtures  /subdir",
+			expectErr: true,
+		},
+		{
+			name:      "trailing whitespace + whitespace in middle of path",
+			path:      "../../test/  fixtures  /subdir  ",
+			expectErr: true,
+		},
+		{
+			name:      "leading and trailing whitespace + whitespace in middle of path",
+			path:      "  ../../test/  fixtures  /subdir  ",
+			expectErr: true,
+		},
 	}
 
 	for _, tt := range tests {
@@ -280,12 +301,18 @@ func Test_FileFinderPathWithLeadingAndTrailingWhitespaces(t *testing.T) {
 
 			files, err := fsFinder.Find()
 
-			if len(files) < 1 {
-				t.Errorf("Unable to find file")
-			}
+			if tt.expectErr {
+				if err == nil {
+					t.Errorf("Error should be thrown for bad path")
+				}
+			} else {
+				if len(files) < 1 {
+					t.Errorf("Unable to find file")
+				}
 
-			if err != nil {
-				t.Errorf("Unable to find file")
+				if err != nil {
+					t.Errorf("Unable to find file")
+				}
 			}
 		})
 	}

--- a/pkg/finder/finder_test.go
+++ b/pkg/finder/finder_test.go
@@ -249,6 +249,48 @@ func Test_FileFinderBadPath(t *testing.T) {
 	}
 }
 
+func Test_FileFinderPathWithLeadingAndTrailingWhitespaces(t *testing.T) {
+	tests := []struct {
+		name string
+		path string
+	}{
+		{
+			name: "no whitespace",
+			path: "../../test/fixtures/subdir",
+		},
+		{
+			name: "leading whitespace",
+			path: "  ../../test/fixtures/subdir",
+		},
+		{
+			name: "trailing whitespace",
+			path: "../../test/fixtures/subdir  ",
+		},
+		{
+			name: "leading and trailing whitespace",
+			path: "  ../../test/fixtures/subdir  ",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fsFinder := FileSystemFinderInit(
+				WithPathRoots(tt.path),
+			)
+
+			files, err := fsFinder.Find()
+
+			if len(files) < 1 {
+				t.Errorf("Unable to find file")
+			}
+
+			if err != nil {
+				t.Errorf("Unable to find file")
+			}
+		})
+	}
+}
+
 func Benchmark_Finder(b *testing.B) {
 	fsFinder := FileSystemFinderInit(
 		WithPathRoots("../../test/fixtures/"),

--- a/pkg/finder/fsfinder.go
+++ b/pkg/finder/fsfinder.go
@@ -81,7 +81,9 @@ func (fsf FileSystemFinder) Find() ([]FileMetadata, error) {
 	seen := make(map[string]struct{}, 0)
 	uniqueMatches := make([]FileMetadata, 0)
 	for _, pathRoot := range fsf.PathRoots {
-		matches, err := fsf.findOne(pathRoot, seen)
+		// remove all leading and trailing whitespace
+		trimmedPathRoot := strings.TrimSpace(pathRoot)
+		matches, err := fsf.findOne(trimmedPathRoot, seen)
 		if err != nil {
 			return nil, err
 		}

